envi/__init__.py

```diff
@@ -156,6 +156,26 @@ def archGetRegisterGroups(self):
         allr = [rname for rname in regctx.getRegisterNames()]
         return [ ('all', allr), ]
 
+    def archModifyFuncAddr(self, va, info):
+        '''
+        Can modify the VA and context based on architecture-specific info.
+        Default: return the same va, info
+
+        This hook allows an architecture to correct VA and Architecture, such 
+        as is necessary for ARM/Thumb.
+        '''
+        return va, info
```

Comment:
if we're using a dict, should we even return it?  we can modify it and not have to return it.  it's less clear, but faster.  your call.